zerver/views/users.py

```diff
@@ -33,6 +33,17 @@ def deactivate_user_backend(request, user_profile, email):
         return json_error(_('No such user'))
     return _deactivate_user_profile_backend(request, user_profile, target)
 
+def deactivate_user_own_backend(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
+    # print(get_admin_users)
+    admins = set(user_profile.realm.get_admin_users())
+
+    if user_profile.can_admin_user(user_profile) and len(admins) == 1:
```

Comment:
I'd replace the first part of this check with just `if user_profile.is_realm_admin`; it's a bit simpler and will be more robust if we change to a model where users are allowed to "admin" themselves.